dissect/target/plugins/os/unix/log/journal.py

```diff
@@ -0,0 +1,385 @@
+                        yield entry_object_offset
+
+                offset = entry_array_object.next_entry_array_offset
+
+    def decode_value(self, value: bytes) -> tuple[str, str]:
+        value = value.decode().strip()
+
+        # Strip leading underscores part of the field name
+        value = value.lstrip("_")
+
+        key, value = value.split("=", 1)
+        key = key.lower()
+
+        return key, value
+
+    def __iter__(self) -> Iterator[Instance]:
+        "Iterate over the entry objects to read payloads."
+
+        for offset in self.entry_object_offsets():
+            self.fh.seek(offset)
+            entry = c_journal.EntryObject(self.fh)
+
+            event = {}
+            event["ts"] = ts.from_unix_us(entry.realtime)
+
+            for item in entry.items:
+                try:
+                    self.fh.seek(item.object_offset)
+
+                    object = c_journal.ObjectHeader(self.fh)
+
+                    if object.type == c_journal.ObjectType.OBJECT_DATA:
+                        self.fh.seek(item.object_offset)
+
+                        data_object = c_journal.DataObject(self.fh)
+
+                        data = data_object.payload
+
+                        if not data:
+                            # If the payload is empty
+                            continue
+                        elif data_object.flags == c_journal.ObjectFlag.OBJECT_COMPRESSED_XZ:
+                            data = lzma.decompress(data)
+                        elif data_object.flags == c_journal.ObjectFlag.OBJECT_COMPRESSED_LZ4:
+                            data = lz4.decompress(data[8:])
+                        elif data_object.flags == c_journal.ObjectFlag.OBJECT_COMPRESSED_ZSTD:
+                            data = zstandard.decompress(data)
+
+                        key, value = self.decode_value(data)
+                        event[key] = value
+
+                except Exception:
+                    continue
+
+            yield event
+
+
+class JournalPlugin(Plugin):
+    JOURNAL_GLOBS = ["/var/log/journal/*", "/run/systemd/journal/*"]
+    JOURNAL_SIGNATURE = "LPKSHHRH"
+
+    def __init__(self, target: Target):
+        super().__init__(target)
+        self.journal_paths = []
+
+        for glob in self.JOURNAL_GLOBS:
+            for journal_path in self.target.fs.glob(glob):
+                self.journal_paths.append(journal_path)
+
+    def check_compatible(self) -> bool:
+        return bool(len(self.journal_paths))
+
+    @export(record=JournalRecord)
+    def journal(self) -> Iterator[JournalRecord]:
+        """Return the content of Systemd Journal log files.
+
+        References:
+            - https://wiki.archlinux.org/title/Systemd/Journal
+            - https://github.com/systemd/systemd/blob/9203abf79f1d05fdef9b039e7addf9fc5a27752d/man/systemd.journal-fields.xml
+        """  # noqa: E501
+
+        for _path in self.journal_paths:
+            for f in self.target.fs.listdir_ext(_path):
```

Comment:
This gives an error on items like:

/run/systemd/journal/flushed

(at least on my system)


maybe try:

`filter(lambda x: self.target.fs.is_dir(x), self.journal_paths)` ?

Reply:
Thi issue does not seem to be resolved.

Reply:
I tried to reproduce the error for you but I now get an EOError on /run/systemd/journal/streams/8:356847. Point is, there seem to be all kinds of objects in the journal folders, maybe we need to test if it's a file first? Maybe even look for the .journal extension?

Reply:
I have added a filter for known extensions

Reply:
I have added that path to the TODO